ecc_enclave/enclave/enclave.cpp

```diff
@@ -229,11 +74,16 @@ int ecall_cc_invoke(const uint8_t* signed_proposal_proto_bytes,
                     cleartext_cc_request.input.args[i]->size));
         }
 
-        // the dynamic memory in the message is release at the end
+        // the dynamic memory in the message is released at the end
     }
 
-    ret = invoke(response, response_len, &response_len_out, &ctx);
-    COND2ERR(ret != 0);
+    invoke_ret = invoke(response, response_len, &response_len_out, &ctx);
+    // invoke_ret is not checked
+
+    // TODO double check or rethink if it is appropriate for a chaincode
+    // to return an error and still forward the response
+    // in particular: should the enclave sign a response? and the rwset? could the tx be committed
+    // though it failed?
```

Comment:
Invoke_ret should be handled when we add the fabric response object (in status field) instead of just using the response byte array, At that time we might also add additional out params to invoke so the chaincode can (as in standard fabric) also pass an error message back ...

Reply:
seems appropriate, we have the todo there.